sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -655,6 +655,30 @@ object TypeCoercion {
       case NaNvl(l, r) if l.dataType == FloatType && r.dataType == DoubleType =>
         NaNvl(Cast(l, DoubleType), r)
       case NaNvl(l, r) if r.dataType == NullType => NaNvl(l, Cast(r, l.dataType))
+
+      case a @ ApproximatePercentile(child, pe @ NumericType(), _, _, _) =>
```

Comment:
we can still check if the `ae` is Integral type, we can give up type coercion earlier if it's not.